pandas/core/frame.py

```diff
@@ -1909,41 +1909,73 @@ def to_dict(self, orient: str = "dict", into=dict):
             elif orient.startswith("i"):
                 orient = "index"
 
+        object_dtype_cols = {
+            col for col, dtype in self.dtypes.items() if is_object_dtype(dtype)
+        }
+        are_all_object_dtype_cols = len(object_dtype_cols) == len(self.dtypes)
         if orient == "dict":
             return into_c((k, v.to_dict(into)) for k, v in self.items())
 
         elif orient == "list":
             return into_c(
-                (k, list(map(maybe_box_native, v.tolist()))) for k, v in self.items()
+                (
+                    k,
+                    list(map(maybe_box_native, v.tolist()))
+                    if k in object_dtype_cols
+                    else v.tolist(),
+                )
+                for k, v in self.items()
             )
 
         elif orient == "split":
+            if are_all_object_dtype_cols:
+                data = [
+                    list(map(maybe_box_native, t))
+                    for t in self.itertuples(index=False, name=None)
+                ]
+            elif object_dtype_cols:
+                # A number of ways were tried here, this solution proved to be the
+                # most optimal in general
```

Comment:
This looks more like a PR comment than a code comment; I don't think it helps the reader in the future. Can you remove.

Reply:
@rhshadrach sure, my thinking was that it's a bit strange to see a list built up and then changed directly after when it could be done in 1 go (but the performance is much better this way), hence the comment, but happy to remove if you think it doesn't help anyone.

Reply:
I see; perhaps "Apply maybe_box_naive after list comprehension for perf" would be more direct? I don't _think_ you're trying to say that this is the fastest way, and that no one else should try to find more performant methods, but that's how I read the comment.